napari/layers/tracks/tracks.py

```diff
@@ -574,17 +574,40 @@ def _norm(p):
         # actually set the vertex colors
         self._track_colors = colormap.map(vertex_properties)
 
-    @property
-    def track_connex(self) -> np.ndarray:
-        """vertex connections for drawing track lines"""
-        return self._manager.track_connex
-
     @property
     def track_colors(self) -> np.ndarray:
         """return the vertex colors according to the currently selected
         property"""
         return self._track_colors
 
+    @track_colors.setter
+    def track_colors(self, colors: np.ndarray) -> None:
+        """update vertex colors given a array of colors or a single colors,
+        colors must be in rgb and between 0 and 1
+        """
+        if colors.ndim != 1 and colors.ndim != 2:
+            raise ValueError(
+                f'`colors` must be 1 or 2-dimensional. Found {colors.ndim} dims.'
+            )
+
+        if colors.ndim == 1:
+            colors = np.tile(colors, (len(self.data), 1))
+
+        if len(colors) != len(self.data):
+            raise ValueError(
+                f'`colors` must match the number of track vertices. Found {len(colors)}, expected {len(self.data)}'
+            )
+
+        if colors.shape[1] == 3:
+            colors = np.append(colors, np.ones((len(colors), 1)), axis=1)
```

Comment:
We have some util functions for color handling.

https://github.com/napari/napari/blob/dbf0fbc62e9bdd974cddfdf9be49085918992cb5/napari/layers/utils/color_transformations.py#L17-L37

https://github.com/napari/napari/blob/dbf0fbc62e9bdd974cddfdf9be49085918992cb5/napari/layers/utils/color_transformations.py#L98-L121

You can use them together like this:

```python
n_colors = len(self.data)

# transform the color into our standard RGBA array
standardized_colors = transform_color_with_defaults(
    num_entries=n_colors,
    colors=colors,
    default='white'
)

# broadcast the colors to the number of data elements
broadcasted_colors = normalize_and_broadcast_colors(
    num_entries=n_colors,
    colors=standardized_colors
)

Reply:
Thanks, very useful, I updated the PR to use this.

Reply:
Great!